src/CoreWCF.MSMQ/src/CoreWCF/Channels/DeadLetterQueueSender.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading.Tasks;
+using MSMQ.Messaging;
+
+namespace CoreWCF.Channels
+{
+    internal class DeadLetterQueueSender : IDeadLetterQueueMsmqSender
+    {
+        public async Task Send(PipeReader message, Uri endpoint)
+        {
+            string nativeQueueName = MsmqQueueNameConverter.GetMsmqFormatQueueName(endpoint);
+            MemoryStream memStream = await ConvertToStream(message);
+            var queue = new MessageQueue(nativeQueueName);
+            var messageForQueue = new MSMQ.Messaging.Message { BodyStream = memStream };
```

Comment:
Do we validate the queue exists before we get to executing this code?

Reply:
Yes, sure